src/Sentry.Unity/SentryOptionsUtility.cs

```diff
@@ -1,42 +1,62 @@
 using Sentry.Extensibility;
 using Sentry.Unity.Integrations;
+using UnityEngine;
 
 namespace Sentry.Unity
 {
     internal static class SentryOptionsUtility
     {
         public static void SetDefaults(SentryUnityOptions options, IApplication? application = null)
         {
+            options.TryAttachLogger();
+
             application ??= ApplicationAdapter.Instance;
 
             SetRelease(options, application);
             SetEnvironment(options, application);
             SetCacheDirectoryPath(options, application);
         }
 
+        public static void LogOptions(SentryUnityOptions options)
+        {
+            var logger = options.DiagnosticLogger;
+            if (logger == null)
+            {
+                return;
+            }
+
+            LogOption(logger, "Release", options.Release);
+            LogOption(logger, "Environment", options.Environment);
```

Comment:
Could this be misleading?

We're only setting if the value isn't set yet, right? The log message is 'setting X to Y'

Reply:
My thinking is: We're setting lots of things by default because we expect there to be a value. So if the user overrides the value with empty it should look odd. 
But we could log it explicitly like with "offline caching disabled"?